specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,959 @@
+
+It is less likely that differences will arise between unbundle and unpack. However, as seen with the existing makemsix tool,
+as long as pack and bundle are separate operations it may not be especially useful for unbundle and unpack to be joined. Unpacking of
+all bundled packages makes sense for a command line tool where an IT admin may be typing in individual commands each time and so
+there's a good reason to avoid having to run multiple commands. In cases where the api is actually being used programattically, 
+to do something like extract all packages and update every manifest with a different version number, there is no avoiding use of filesystem
+apis to find each unpacked package. In such a case merging the unbundle and unpack command into one api just adds hidden complexity into the api
+in the form of either requiring more configuration options or hiding decisions about default choices like folder naming decisions in the api documentation.
+
+For these reasons pack, bundle, unpack, unbundle have been left as separate commands. 
+
+MSIXMgr defines an unpack command which has fairly different usage from the makeappx unpack command.
+The msixmgr unpack command merges creation of an image file and adding packages to that image file into one command.
+The msixmgr command line supports adding all packages in a folder to an image at once. The CIM file format it supports always expands to fit those apps,
+but msixmgr does not currently support expandable vhdx files. This means that the caller needs to know up front the size of all unpacked packages when
+creating vhds, which will be difficult to know without either unpacking them first or reading each package's block map to get file sizes.
+This api proposes getting rid of that requirement and having the api implementation itself determine the size of all packages being added in 
+order to create an image of the correct size. This behavior is more consistent with other types of package creation and eliminates the api differences
+between CIM files and VHD\VHDX files. 
+
+Since the api surface supports creation of the CIM, VHD, VHDX file as well as creation of msix and msixbundle files, callers may
+find it strange to need to first pack their directory to an msix, and then create the vhdx from the msix, rather than just
+creating the vhdx directly. However, signing is not integrated into any of these tools, and is a separate step that
+is done after packaging is complete. The Appx SIP knows how to sign packages, but not vhdx and other mounted directories. The result
+is that creating a vhdx with an appx signature file (appxsignature.p7x) requires first creating the package, then signing it, then
+creating the vhdx. Allowing creation of a CIM, VHD, or VHDX file directly from a directory seems likely to cause confusion for 
+developers who may not realize that they won't be able to sign the package in the image. This api also proposes always applying acls
```

Comment:
The first one - no direct cim\vhd\vhdx from a package layout directory. My claim is that since other apis on the system do not recognize cim\vhd\vhdx as a package in and of itself, rather than just a container for packages, that it will be confusing to have this api treat those formats as if they're themselves a type of package just because it's possible to create a cim\vhd\vhdx with only one package. A layout directory is packed, then signed, then added to a cim\vhd\vhdx in separate steps.